docs/testplan/Smartswitch-test-plan.md

```diff
@@ -0,0 +1,966 @@
+| 1.2 | Check modified ChassisClass APIs for SmartSwitch       |  To verify the existing ChassisClass APIs that undergo minor changes with the addition of SmartSwitch| |
+| 1.3 | Check DpuModule APIs for SmartSwitch       |  To verify the newly implemented  DpuModule APIs for SmartSwitch| |
+| 1.4 | Check modified ModuleClass APIs for SmartSwitch       |  To verify the existing ModuleClass APIs that undergo minor changes with the addition of SmartSwitch| 
+
+## API Test Cases 
+
+### 1.1 Check SmartSwitch specific ChassisClass APIs
+
+#### Steps
+ * Execute the following APIs on SmartSwitch
+ * get_dpu_id(self, name):
+    * Provide name (Example: DPU0 - Get it from platform.json file)
+    * This API should return an integer from 1-8 (check it against platform.json)
+ * is_smartswitch(self):
+    * This API should return True
+ * get_module_dpu_data_port(self, index):
+    * It will return a dict as shown below.
+
+
+#### Verify in
+ * Switch
+
+#### Sample Output
+```
+On Switch:
+    get_dpu_id(self, DPU3)
+    Output: 4
+    is_smartswitch(self):
+    Output: True
+    get_module_dpu_data_port(self, DPU0):
+    Output: {
+                "interface": {"Ethernet224": "Ethernet0"}
+            }
+```
+#### Pass/Fail Criteria
+ *  The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output".
+
+
+### 1.2 Check modified ChassisClass APIs for SmartSwitch
+
+#### Steps
+ * is_modular_chassis(self):
+    * Should return False
+ * get_num_modules(self):
+    * Should return number of DPUs
+ * get_module(self, index):
+    * Make sure for each index this API returns an object and has some content and not None
+    * Check that the object's class is inherited from the ModuleBase class
+ * get_all_modules(self):
+    * This should return a list of items
+ * get_module_index(self, module_name):
+    * Given the module name say “DPU0” should return the index of it “1”
+
+
+#### Verify in
+ * Switch
+
+#### Sample Output
+```
+On Switch:
+    is_modular_chassis(self):
+    Output: False
+    get_num_modules(self):
+    Output: number of DPUs
+    get_module(self, DPU0):
+    Output: DPU0 object
+    get_all_modules(self):
+    Output: list of objects (one per DPU + 1 switch object)
+    get_module_index(self, DPU0):
+    Output: could be any value from 0 to modules count -1
+```
+#### Pass/Fail Criteria
+ * The test result is a pass if the return value matches the expected value as shown in the "steps" and "Sample Output"
+
+
+### 1.3 Check DpuModule APIs for SmartSwitch
+
+#### Steps
+ * get_dpu_id(self):
+    * Should return ID of the DpuModule Ex: 1 on DPU0
+* get_reboot_cause(self):
+    * Reboot the module and then execute the "show reboot-cause ..." CLIs
+    * Verify the output string shows the correct Time and Cause
+    * Limit the testing to software reboot
+ * get_state_info(self):
+    * This should return an object
```

Comment:
Can you please show an example of the object? I don't recall it is described in the HLD

Reply:
Sample output is added.